client/src/api/rooms.js

```
@@ -0,0 +1,39 @@
+import axios from 'axios';
+
+const BASE_URL = 'http://localhost:5000';
+
+export async function fetchRooms() {
+  try {
+    const res = await axios.get(`${BASE_URL}/rooms`);
+    return res.data;
+  } catch (err) {
+    console.error('Error fetching rooms:', err);
+  }
```

fetchRooms swallows errors in its catch block. Let the error propagate (rethrow) so consumers can handle a failed fetch. Also consider making the console message more specific (e.g., 'Error fetching rooms').